### PR TITLE
Update algorithms used for heart rate calculation

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
I had to remove a number of the unit tests that check "the math" at interim steps in the processing b/c I no longer have valid data for those stages of the processing. 

This version is matching against:
https://github.com/itismeghasyam/mHealthTools/commit/503f6bdcb77d4a8ad625ddf3dae3949057936aab
https://github.com/itismeghasyam/CRF_validation_analysis/commit/aef33b81132d302542abb1dc398b0091a9fbe2c6

### CODE REVIEW

File: CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/CRFHeartRateSampleProcessor.swift [hunk 1]
```diff
@@ -458,12 +463,17 @@ internal class CRFHeartRateSampleProcessor {
         // it is reproduced here for testing purposes.
         let drop = dropSeconds > 0 ? dropSeconds * samplingRate - 1 : 0
         let x = input.dropFirst(drop).map({ $0.isFinite ? $0 : 0 })
+        // get minimum value in the input window and offset each signale frame by the minimum value observed in the window
+        let minValue = x.min()
+        let minnedOutX = x.map({$0 - minValue!})
```
Comment (nategbrown9) [inline CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/CRFHeartRateSampleProcessor.swift]: Do we need to worry about the forced unwrap of minValue?  Should it be:
`let minValue = x.min() ?? 0.0`
`let minnedOutX = x.map({$0 - minValue})`

**Review by nategbrown9 — APPROVED** [review]
One comment, otherwise looks good.

File: CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/CRFHeartRateSampleProcessor.swift (hunk 1 above)
Comment (syoung-smallwisdom) [reply CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/CRFHeartRateSampleProcessor.swift]: Not really, though it has less code smell. :)

File: CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/CRFHeartRateSampleProcessor.swift [hunk 2]
```diff
@@ -170,7 +173,9 @@ internal class CRFHeartRateSampleProcessor {
             
             // Need to keep 2 extra seconds due to filtering lopping off the first 2 seconds of data.
             let meanOrder = self.meanFilterOrder(for: roundedRate)
-            let windowLength = Int(CRFHeartRateWindowSeconds + 2) * roundedRate + meanOrder
+            
+            //set window before drop to 15 seconds
```
Comment (Erin-Mounts) [inline CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/CRFHeartRateSampleProcessor.swift]:
```suggestion
            // set window before drop to 15 seconds
```

File: CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/CRFHeartRateSampleProcessor.swift [hunk 3]
```diff
@@ -170,7 +173,9 @@ internal class CRFHeartRateSampleProcessor {
...
+            let windowLength = (Int(CRFHeartRateWindowSeconds) + 2*CRFHeartRateFilterDropSeconds) * roundedRate + meanOrder
```
Comment (Erin-Mounts) [inline CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/CRFHeartRateSampleProcessor.swift]:
```suggestion
            let windowLength = (Int(CRFHeartRateWindowSeconds) + 2 * CRFHeartRateFilterDropSeconds) * roundedRate + meanOrder
```

File: CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/CRFHeartRateSampleProcessor.swift [hunk 4]
```diff
@@ -458,12 +463,17 @@ internal class CRFHeartRateSampleProcessor {
         // it is reproduced here for testing purposes.
         let drop = dropSeconds > 0 ? dropSeconds * samplingRate - 1 : 0
         let x = input.dropFirst(drop).map({ $0.isFinite ? $0 : 0 })
+        // get minimum value in the input window and offset each signale frame by the minimum value observed in the window
```
Comment (Erin-Mounts) [inline CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/CRFHeartRateSampleProcessor.swift]:
```suggestion
        // get minimum value in the input window and offset each signal frame by the minimum value observed in the window
```

**Review by Erin-Mounts — APPROVED** [review]
a few nits, otherwise looks good

**Review by DanWebster — APPROVED** [review]
Works great, successful completion rate for me is above 95% across all tasks

**Pull request merged**